File: app/src/main/java/com/veganbeauty/app/data/local/dao/UserDao.java
package com.veganbeauty.app.data.local.dao;

import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;
import com.veganbeauty.app.data.local.entities.UserEntity;
import kotlinx.coroutines.flow.Flow;
import java.util.List;

@Dao
public interface UserDao {
    @Query("SELECT * FROM users")
    Flow<List<UserEntity>> getAllUsers();

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    long insertUserSync(UserEntity user);

    @androidx.annotation.Nullable
    @Query("SELECT * FROM users WHERE user_id = :userId LIMIT 1")
    UserEntity getUserByIdSync(String userId);

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    long[] insertUsersSync(List<UserEntity> users);

    @androidx.annotation.Nullable
    @Query("SELECT * FROM users WHERE email = :email AND password = :password LIMIT 1")
    UserEntity getUserByEmailAndPasswordSync(String email, String password);

    @androidx.annotation.Nullable
    @Query("SELECT * FROM users WHERE phone = :phone AND password = :password LIMIT 1")
    UserEntity getUserByPhoneAndPasswordSync(String phone, String password);

    @androidx.annotation.Nullable
    @Query("SELECT * FROM users WHERE email = :email LIMIT 1")
    UserEntity getUserByEmailSync(String email);

    @androidx.annotation.Nullable
    @Query("SELECT * FROM users WHERE phone = :phone LIMIT 1")
    UserEntity getUserByPhoneSync(String phone);

    @Query("DELETE FROM users WHERE username = :username")
    int deleteUserByUsernameSync(String username);

    @Query("UPDATE users SET password = :newPassword WHERE email = :email")
    int updatePasswordByEmailSync(String email, String newPassword);

    @Query("UPDATE users SET password = :newPassword WHERE phone = :phone")
    int updatePasswordByPhoneSync(String phone, String newPassword);
}

File: app/src/main/java/com/veganbeauty/app/data/local/dao/UserDao.kt
package com.veganbeauty.app.data.local.dao

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query
import com.veganbeauty.app.data.local.entities.UserEntity
import kotlinx.coroutines.flow.Flow

@Dao
interface UserDao {
    @Query("SELECT * FROM users")
    fun getAllUsers(): Flow<List<UserEntity>>

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    suspend fun insertUser(user: UserEntity): Long

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insertUserSync(user: UserEntity): Long

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    suspend fun insertUsers(users: List<UserEntity>): List<Long>

    @Query("SELECT * FROM users WHERE email = :email AND password = :password LIMIT 1")
    suspend fun getUserByEmailAndPassword(email: String, password: String): UserEntity?

    @Query("SELECT * FROM users WHERE phone = :phone AND password = :password LIMIT 1")
    suspend fun getUserByPhoneAndPassword(phone: String, password: String): UserEntity?

    @Query("SELECT * FROM users WHERE email = :email LIMIT 1")
    suspend fun getUserByEmail(email: String): UserEntity?

    @Query("SELECT * FROM users WHERE phone = :phone LIMIT 1")
    suspend fun getUserByPhone(phone: String): UserEntity?

    @Query("DELETE FROM users WHERE username = :username")
    fun deleteUserByUsernameSync(username: String)

    @Query("SELECT * FROM users WHERE email = :email LIMIT 1")
    fun getUserByEmailSync(email: String): UserEntity?

    @Query("SELECT * FROM users WHERE phone = :phone LIMIT 1")
    fun getUserByPhoneSync(phone: String): UserEntity?

    @Query("UPDATE users SET password = :newPassword WHERE email = :email")
    fun updatePasswordByEmailSync(email: String, newPassword: String)

    @Query("UPDATE users SET password = :newPassword WHERE phone = :phone")
    fun updatePasswordByPhoneSync(phone: String, newPassword: String)
}
